graph-gateway/src/network/subgraph.rs

```diff
@@ -0,0 +1,249 @@
+            ) {{
+                id
+                {}
+                versions(orderBy: version, orderDirection: desc) {{
+                    version
+                    subgraphDeployment {{
+                        ipfsHash
+                        manifest {{
+                            network
+                            startBlock
+                        }}
+                        indexerAllocations(
+                            first: 100
+                            orderBy: allocatedTokens, orderDirection: desc
+                            where: {{ status: Active }}
+                        ) {{
+                            id
+                            indexer {{
+                                id
+                            }}
+                        }}
+                        {}
+                    }}
+                }}
+            }}
+        "#,
+            self.l2_transfer_support
+                .then_some("")
+                .unwrap_or("active: true"),
+            self.l2_transfer_support.then_some("idOnL2").unwrap_or(""),
+            self.l2_transfer_support
+                .then_some("transferredToL2")
+                .unwrap_or(""),
+        );
+
+        self.client
+            .paginated_query(query, 200)
+            .await
+            .map_err(|err| anyhow::anyhow!(err))
+    }
+
+    /// Fetch the list of indexers from the network subgraph.
+    ///
+    /// Performs a paginated query to fetch the latest state of the network subgraph:
+    ///
+    /// > Get all indexers with at least one active allocation, with all its active allocations
+    /// > ordered by `allocatedTokens` in descending order; laying the largest allocation first.
+    pub async fn fetch_indexers(&mut self) -> anyhow::Result<Vec<types::fetch_indexers::Indexer>> {
+        // The following query response must fulfill the following assumptions:
+        // - Perform a paginated query based on the indexer's ID. All indexers are ordered by ID.
+        //   The next query must start from the last indexer ID fetched in the previous query.
+        // - Get all indexers with at least one active allocation.
+        // - To get the largest allocation, the query must fetch all active allocations ordered by
+        //  `allocatedTokens` in descending order; the largest allocation must be the first one.
+        //
+        // ref#: d260724b-a445-4842-964e-fb95062c119d
+        let query = r#"
+            indexers(
+                block: $block
+                orderBy: id, orderDirection: asc
+                first: $first
+                where: { 
+                    id_gt: $last
+                    allocationCount_gt: 0
+                }
+            ) {
+                id
+                url
+                stakedTokens
+                allocations(
+                    first: 100,
+                    orderBy: allocatedTokens, orderDirection: desc
+                    where: { status: Active }
+                ) {
```

Comment:
This filter only makes sense if it's the first n indexer allocations per subgraph deployment. When querying for indexers, many will have more than 100 allocations (like upgrade indexer). So this would cause problems.